File: python/IoT platform/Mobius_APIs_Python/onem2m_get_la_img.py
# Created by J. Yun, SCH Univ., yun@sch.ac.kr

import requests
import base64

url = 'http://203.253.128.161:7579/Mobius/sch19999999/img/la'
headers = {'Accept':'application/json',
			'X-M2M-RI':'12345',
			'X-M2M-Origin':'SOrigin'}

r = requests.get(url, headers=headers)

try:
	r.raise_for_status()
	jr = r.json()
	
	print(jr)
	print(jr['m2m:cin']['con'])

	with open("img_to_save.png", "wb") as f:
		f.write(base64.b64decode(jr['m2m:cin']['con']))

except Exception as exc:
	print('There was a problem: %s' % (exc))

File: nodejs/onem2m_get_la_img.js
// Created by J. Yun, SCH Univ., yun@sch.ac.kr
// Need to install node-fetch
// $ npm i node-fetch --save

const fetch = require("node-fetch");

var myHeaders = new fetch.Headers();
myHeaders.append("Accept", "application/json");
myHeaders.append("X-M2M-RI", "12345");
myHeaders.append("X-M2M-Origin", "justin");

var requestOptions = {
  method: 'GET',
  headers: myHeaders,
  redirect: 'follow'
};

fetch("http://203.253.128.161:7579/Mobius/sch19999999/img/la", requestOptions)
  .then(
    function(response) {
      if (response.status !== 200) {
        console.log('There was a problem. Status Code: ' +
          response.status);
        return;
      }

      response.json().then(function(data) {
        require('fs').writeFile('./img_to_save.png', data['m2m:cin']["con"], {encoding: 'base64'}, function(err) {console.log('File created')});        
      });
    }
  )
  .catch(function(err) {
    console.log('Fetch Error :-S', err);
  });